Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,243 @@
+
+    # Wait for 30s and check for service.log
+    sleep 30
+    [ ! -f "/var/opt/microsoft/dependency-agent/log/service.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log does not exist." && SetTestStateFailed && exit 0
+
+    # Check for service.log.1 file and confirm service.log.2 doesn't exist
+    service_log_time=$(date -r /var/opt/microsoft/dependency-agent/log/service.log +%s)
+    time_limit=$(($service_log_time + 120))
+    while [ ! -f "/var/opt/microsoft/dependency-agent/log/service.log.1" ]
+    do
+        current_time=$(date +%s)
+        if [ $current_time -gt $time_limit]; then
+            SetTestStateFailed
+            return
+        fi
+        sleep 5
+    done
+
+    [ ! -f "/var/opt/microsoft/dependency-agent/log/service.log.1" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log.1 does not exist." && SetTestStateFailed && exit 0
+    [ -f "/var/opt/microsoft/dependency-agent/log/service.log.2" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log.2 exist." && SetTestStateFailed && exit 0
+    [ ! -c "/dev/msda" ] && LogErr "/dev/msda does not exist." && SetTestStateFailed && exit 0
+
+    # Check for "driver setup status=0" and "starting the dependency agent" in service.log.1 and service.log respectively
+    ! grep -iq "driver setup status=0" /var/opt/microsoft/dependency-agent/log/service.log.1 && LogErr "driver setup status=0 not found" && SetTestStateFailed && exit 0
+    ! grep -iq "starting the dependency agent" /var/opt/microsoft/dependency-agent/log/service.log && LogErr "starting the dependency agent not found" && SetTestStateFailed && exit 0
+
+    # Wait for DA to be running and verify by checking for MicrosoftDependencyAgent.log
+    sleep 60
+    [ ! -f "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log does not exist." && SetTestStateFailed && exit 0
+    verify_da_pid
+
+    # Wait for events and verify by checking for /var/opt/microsoft/dependency-agent/storage/*.bb files
+    sleep 120
+    ! ls /var/opt/microsoft/dependency-agent/storage/*.bb && LogErr "/var/opt/microsoft/dependency-agent/storage/*.bb does not exist." && SetTestStateFailed && exit 0
+
+    # Wait for more events and verify by checking for /var/opt/microsoft/dependency-agent/storage/*.hb files
+    sleep 90
+    ! ls /var/opt/microsoft/dependency-agent/storage/*.hb && LogErr "/var/opt/microsoft/dependency-agent/storage/*.hb does not exist." && SetTestStateFailed && exit 0
+    verify_da_pid
+
+    dmesg_error=$(dmesg | fgrep -e BUG: -e "Modules linked in:" -e "Call Trace:​" -A 20)
+    [ ! -z "$dmesg_error" ] && LogErr "$dmesg_error" && SetTestStateFailed && exit 0
+
+    verify_uninstall_da
+
+    LogMsg "Enable/Disable DA tests passed successfully"
+}
+
+function test_case_install_uninstall_da(){
+    LogMsg "Starting Test Case 1: Install/Uninstall DA"
+    check_prereqs
```

Comment:
Checking the prereqs is not part of either test case. They should be done before either test case.
Side note: I've worked with experienced, talented test engineers and they taught me that each test case should stand alone and there should never be an order dependency between them. So, for example here, if test case 2 was run **before** test case 1, it could be failing because the pre-requisites weren't met and you'd be left scratching your head why it was failing.

Reply:
Done